src/modbus-tcp.c

```diff
@@ -729,7 +729,9 @@ static int _modbus_tcp_select(modbus_t *ctx, fd_set *rset, struct timeval *tv, i
 }
 
 static void _modbus_tcp_free(modbus_t *ctx) {
-    free(ctx->backend_data);
+    if (ctx->backend_data) {
+        free(ctx->backend_data);
+    }
```

Comment:
ctx is always valid here, since there is a check in `modbus_free` already.

Thanks for your prompt review. Will update PR in a few minutes...